File: gdx2d-demoDesktop/src/main/java/ch/hevs/gdx2d/demos/shaders/simple/DemoShaderSimple.java
package ch.hevs.gdx2d.demos.shaders.simple

import ch.hevs.gdx2d.desktop.PortableApplication
import ch.hevs.gdx2d.lib.GdxGraphics

/**
 * Shows how to interwind shaders and normal GDX operations
 *
 * @author Pierre-André Mudry (mui)
 * @version 0.1
 */
class DemoShaderSimple : PortableApplication() {

    private val time = 0f

    override fun onInit() {
        this.setTitle("Simple shader demo, mui 2013")
    }

    override fun onGraphicRender(g: GdxGraphics) {
        if (g.shaderRenderer == null) {
            g.setShader("shader/bicolor.fp")
        }

        g.clear()

        // Draws the shader
        g.drawShader()

        g.drawFPS()
        g.drawSchoolLogo()
    }

    companion object {

        @JvmStatic
        fun main(args: Array<String>) {
            DemoShaderSimple()
        }
    }
}

File: gdx2d-demoDesktop/src/main/scala/ch/hevs/gdx2d/demos/shaders/simple/DemoShaderSimple.scala
package ch.hevs.gdx2d.demos.shaders.simple

import ch.hevs.gdx2d.desktop.DesktopApplication
import ch.hevs.gdx2d.lib.GdxGraphics

/**
 * Shows how to interwind shaders and normal GDX operations.
 *
 * @author Pierre-André Mudry (mui)
 * @version 2.0
 */
class DemoShaderSimple extends DesktopApplication {

  override def onInit(): Unit = {
    setTitle("Simple shader demo, mui 2013")
  }

  override def onGraphicRender(g: GdxGraphics): Unit = {
    if (g.getShaderRenderer == null) {
      g.setShader("shader/bicolor.fp")
    }

    g.clear()
    g.drawShader()
    g.drawFPS()
    g.drawSchoolLogo()
  }
}

object DemoShaderSimple {
  def main(args: Array[String]): Unit = {
    new DemoShaderSimple().launch()
  }
}
